applications/processor/session-filter/src/main/java/com/csgroup/s2pdgs/sessionfilter/SessionFilterProcessor.java

```diff
@@ -1,105 +1,253 @@
                 if (!t) {
-                    log.error("Product {} of type {} not updated to catalog catalog", product.getId(), product.getContentType());
+                    log.error("Session {} not added to catalog catalog", product.getId());
                 }
             })
-            .map(t -> t ? Option.of(product) : Option.<StationProduct>none())
-            .doOnError(t -> log.error("Product {} of type {} not updated to catalog catalog because of error {}", product.getId(), product.getContentType(), t))
-            .onErrorResume(t -> Mono.just(Option.none()))
-            .switchIfEmpty(Mono.just(Option.none()))
-            .block();
+            .map(t -> t ? product : null);
+    }
+
+    private enum SessionStatus {
+        COMPLETE,
+        NOT_COMPLETE
+    }
+
+    private SessionStatus isSessionComplete(StationProduct product) throws IOException {
+        var sessionPath = Path.of(product.getLocalPath()).getParent().getParent();
+        var dsibFiles = Files.walk(sessionPath).filter(p -> p.toFile().getName().endsWith("DSIB.xml")).toList();
+        if(dsibFiles.size() < NB_DSIB_PER_SESSION){
+            return SessionStatus.NOT_COMPLETE;
+        }
+        if(dsibFiles.stream().anyMatch(dsib -> areAllRawDownloaded(dsib) == SessionStatus.NOT_COMPLETE)){
+            return SessionStatus.NOT_COMPLETE;
+        }
+
+        return SessionStatus.COMPLETE;
+
+    }
+
+    private SessionStatus areAllRawDownloaded(Path dsib){
+        try (InputStream in = Files.newInputStream(dsib)) {
+            var sib = SessionInformationBlock.from(in);
+
+            var channelPath = dsib.getParent();
+            var channelFiles = Files.list(channelPath).toList();
+            if(!sib.getDsdbList().stream().allMatch(raw -> channelFiles.contains(channelPath.resolve(raw)))){
+                return SessionStatus.NOT_COMPLETE;
+            }
+            long totalDeclaredSize = sib.getDataSize();
+            var rawFilesSize = channelFiles.stream()
+                .filter(e -> sib.getDsdbList().contains(e.toFile().getName()))
+                .mapToLong(this::fileSize)
+                .sum();
+            if (totalDeclaredSize != rawFilesSize){
+                log.warn("Session {} not complete {} bytes expected but {} bytes received",
+                    sib.getSessionID(), totalDeclaredSize, rawFilesSize);
+            }
+            // FIXME do not tell that session is complete if some bytes are missing (corrupted may be)
+            return SessionStatus.COMPLETE;
+        } catch (IOException | JAXBException e) {
+            // FIXME do we attempt a retry or return failure ???
+            log.error("Error checking DSIB completness {}", e.getMessage(), e);
+        }
+        return SessionStatus.NOT_COMPLETE;
+    }
+
+    // FIXME Do not inhibit exception (return failure ???)
+    // FIXME add utility class (duplicate with DownloadDataBlock)
+    private long fileSize(Path p){
+        try{
```

Comment:
Un équivalent avec Vavr:
```suggestion
        return Try.of(() -> Files.size(p))
            .onFailure(e -> log.error("Unable to compute size of file {}: {}", p, e))
            .getOrElse(0L);
```